### ISSUE TITLE
Download fails when behind SSL inspection corporate proxy

### ISSUE THREAD

**Issue by lucastheisen** [puppeteer/puppeteer#1735]
Installation fails due to UNABLE_TO_GET_ISSUER_CERT_LOCALLY.  I am behind a corporate proxy that does SSL inspection (with our own certificate authority), so I am not unused to this type of error.  However, I have configured npm correctly:
```
PS C:\Users\ltheisen\git\facility-report-viewer-ui> npm config get cafile
C:\cygwin64\etc\pki\ca-trust\extracted\pem\tls-ca-bundle.pem
```
Along with the proper proxy settings:
```
PS C:\Users\ltheisen\git\facility-report-viewer-ui> gci env: | findstr proxy
http_proxy                     http://gatekeeper-w.mitre.org:80
https_proxy                    http://gatekeeper-w.mitre.org:80
no_proxy                       .mitre.org,.portal.local,localhost,mm206678-pc,mm206678-pc.mitre.org,127.0.0.1
```
And it works for normal installs.  It seems like your `install.js` respects the `https_proxy` environment variable, but ignores the cafile setting.  I'm not really surprised as it is an npm config and your code is built to work with npm/yarn/..., but I need a way to provide your installer with my CA file.

### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 0.13.0
* Platform / OS version: Windows 10 Enterprise
* URLs (if applicable):
* Node.js version: v9.3.0

**What steps will reproduce the problem?**

_Please include code that reproduces the issue._

1. `npm install --save-dev puppeteer`

**What is the expected result?**
Puppeteer gets installed.

**What happens instead?**
```
PS C:\Users\ltheisen\git\facility-report-viewer-ui> npm install puppeteer --save-dev

> puppeteer@0.13.0 install C:\Users\ltheisen\git\facility-report-viewer-ui\node_modules\puppeteer
> node install.js

ERROR: Failed to download Chromium r515411! Set "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" env variable to skip download.
{ Error: unable to get local issuer certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1042:34)
    at TLSSocket.emit (events.js:159:13)
    at TLSSocket._finishInit (_tls_wrap.js:639:8) code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY' }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! puppeteer@0.13.0 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the puppeteer@0.13.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by Marcel-G** [comment]
Experiencing the same issue.

**Comment by aslushnikov** [comment]
@lucastheisen @Marcel-G I'd appreciate any help with this and will be happy to review a PR.

**Comment by lucastheisen** [comment]
@aslushnikov , It appears your master branch _may_ have already resolved this.  I noticed significant changes from the `0.13.0` version currently released to npm and when I clone, then run using:
```
PS C:\Temp\puppeteer> npm install C:\Users\ltheisen\git\googlechrome-puppeteer\

> puppeteer@1.0.0-rc install C:\Temp\puppeteer\node_modules\puppeteer
> node install.js

npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN worker-loader@1.1.0 requires a peer of webpack@^2.0.0 || ^3.0.0 but none was installed.
npm WARN puppeteer-test@0.0.0 No repository field.

+ puppeteer@1.0.0-rc
updated 1 package in 64.568s
```
It appears to work...  Whereas if I `git checkout v0.13.0` branch it fails in the same fashion as it does from npm:
```
PS C:\Temp\puppeteer> npm install C:\Users\ltheisen\git\googlechrome-puppeteer\

> puppeteer@0.13.0 install C:\Temp\puppeteer\node_modules\puppeteer
> node install.js

ERROR: Failed to download Chromium r515411! Set "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" env variable to skip download.
{ Error: unable to get local issuer certificate
    at TLSSocket.onConnectSecure (_tls_wrap.js:1042:34)
    at TLSSocket.emit (events.js:159:13)
    at TLSSocket._finishInit (_tls_wrap.js:639:8) code: 'UNABLE_TO_GET_ISSUER_CERT_LOCALLY' }
npm WARN worker-loader@1.1.0 requires a peer of webpack@^2.0.0 || ^3.0.0 but none was installed.
npm WARN puppeteer-test@0.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! puppeteer@0.13.0 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the puppeteer@0.13.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\ltheisen\AppData\Roaming\npm-cache\_logs\2018-01-09T20_35_47_271Z-debug.log
```

That said, I am unable to verify using direct npm install from git:
```
PS C:\Temp\puppeteer> npm install git+https://git@github.com/GoogleChrome/puppeteer.git
npm WARN puppeteer-test@0.0.0 No repository field.

npm ERR! path C:\Temp\puppeteer\node_modules\puppeteer\node_modules\ws
npm ERR! code ENOENT
npm ERR! errno -4058
npm ERR! syscall rename
npm ERR! enoent ENOENT: no such file or directory, rename 'C:\Temp\puppeteer\node_modules\puppeteer\node_modules\ws' -> 'C:\Temp\puppeteer\node_modules\ws'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
^CTerminate batch job (Y/N)?
```
So it looks like we will have to wait until you deploy the 1.0.0 release to npm before we can verify.  If you want to comment back at me after the 1.0.0 release, i would be happy to verify from my end...

**Comment by aslushnikov** [comment]
@lucastheisen the tip-of-tree puppeteer is published to npm as puppeteer@next. Could you please give it a try?

```
npm install --save puppeteer@next
```

**Comment by lucastheisen** [comment]
@aslushnikov , it works:
```
PS C:\Temp\puppeteer> npm install --save puppeteer@next

> puppeteer@1.0.0-next.1515463970965 install C:\Temp\puppeteer\node_modules\puppeteer
> node install.js

Downloading Chromium r526987 - 119.9 Mb [====================] 100% 0.0s
Chromium downloaded to C:\Temp\puppeteer\node_modules\puppeteer\.local-chromium\win64-526987
npm notice created a lockfile as package-lock.json. You should commit this file.
+ puppeteer@1.0.0-next.1515463970965
added 41 packages in 13.987s
```
Thank you!

**Comment by SampathKumar27896** [comment]
Hi team,
I am recently trying to download puppeteer on my machine. And I am facing this issue
Installation fails due to `UNABLE_TO_GET_ISSUER_CERT_LOCALLY`. I got to see this discussion. 
when I tried to run the command `npm install --save puppeteer@next` I am facing the below issue. Do we have any idea on this?. 

OS: windows 11
Node js version: 18.16.0

npm ERR! code ETARGET
npm ERR! notarget No matching version found for puppeteer@next.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
